contributing.md

```diff
@@ -0,0 +1,118 @@
+# Contributing to TypeDoc
+
+Thanks for taking the time to contribute! TypeDoc is a volunteer-run project and we couldn't do it without your help.
+
+This document includes a set of guidelines for contributing to TypeDoc. These are guidelines, not rules. If something seems off, please feel free to propose changes to this document in a pull request.
+
+## Table of Contents
+
+1. [How Can I Contribute?](#how-can-i-contribute)
+   - [Bug Reports](#bug-reports)
+   - [Suggestions](#suggestions)
+   - [Code](#code)
+1. [Setup - Git, GitHub, and Node](#setup---git-github-and-node)
+1. [Linting, Building, and Testing](#linting-building-and-testing)
+1. [Pull Requests](#pull-requests)
+1. [Updating Your Branch](#updating-your-branch)
+
+
+## How Can I Contribute?
+
+### Bug Reports
+
+This section guides you through submitting a bug report for TypeDoc. Following these guidelines helps others understand your report and resolve the issue.
+
+Before creating a bug report please check [this list][bugs] list to see if it has already been reported. If the issue is closed, please open a new issue and link to the original issue.
+
+When creating a bug report, explain the problem and include as much additional information as necessary to help maintainers reproduce it. Ideally, provide an example project which highlights the problem.
+
+- **Use a clear and descriptive title** for the issue to identify the problem
+- **Describe your project setup**. The easier it is for maintainers to reproduce your problem, the more likely it is to be fixed quickly.
+- **Explain what you expected to see instead and why**
+
+### Suggestions
+
+This section guides you through submitting an enhancement suggestion for Typedoc.
+
+Before creating a feature request, please check [this list][suggestions] to see if it has already been requested.
+
+When creating an enhancement request, explain your use case and ultimate goal. This will make it possible for contributors to suggest existing alternatives which may already meet your requirements.
+
+- **Use a clear and descriptive title** for the issue to identify the suggestion.
+- **Provide an example where this enhancement would improve TypeDoc**
+- **If possible, list another documentation generator where this feature exists**
+
+### Code
+
+Unsure of where to begin contributing to TypeDoc? You can start by looking through the issues labeled [good-first-issue] and [help-wanted]. Issues labeled with [good-first-issue] should only require changing a few lines of code and a test or two. Issues labeled with [help-wanted] can be considerably more involved and may require changing multiple files.
+
+For instructions on setting up your environment, see the [setup](#setup---git-github-and-node) instructions in this document.
+
+If you have started work on an issue and get stuck or want a second opinion on your implementation feel free to reach out through [Gitter].
```

Comment:
I'm wondering if there is a way to make the gitter channel more visible. It is really helpful to iterate on ideas before a PR is submitted.

Reply:
I've added the Gitter link to the PR section as well, I would have liked to put it under the "How Can I Contribute" section, but struggled to come up with appropriate wording.